plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,265 @@
+			tags["temp_path"] = tempSpace["path"].(string)
+			fields["temp_available"] = tempSpace["size"].(float64)
+		}
+
+		if info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor != nil {
+			swapSpace := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor.(map[string]interface{})
+			fields["swap_available"] = swapSpace["availableSwapSpace"].(float64)
+			fields["swap_total"] = swapSpace["totalSwapSpace"].(float64)
+			fields["memory_available"] = swapSpace["availablePhysicalMemory"].(float64)
+			fields["memory_total"] = swapSpace["totalPhysicalMemory"].(float64)
+		}
+
+		acc.AddFields(measurement, fields, tags)
+
+	}
+
+	return nil
+}
+
+func (j *Jenkins) GetJobsData(jobs []*gojenkins.Job, acc telegraf.Accumulator) error {
+
+	measurement := "jenkins_job"
+
+	for _, job := range jobs {
+
+		jobName := job.GetName()
+
+		for _, filtername := range j.JobFilterName {
+			if filtername == jobName {
+				continue
+			}
+		}
+
+		// ignore if job has no builds
+		if job.Raw.LastBuild.Number < 1 {
+			continue
+		}
+
+		jobLastBuild, err := job.GetLastBuild()
+		if err != nil {
+			log.Printf("E! Error retrieving last build from job %s: %s", jobName, err.Error())
+			return err
+		}
+
+		// ignore if last build is too old
+		if (j.LastbuildFilterInterval != internal.Duration{Duration: 0}) {
+			buildSecAgo := time.Now().Sub(jobLastBuild.GetTimestamp()).Seconds()
+
+			if buildSecAgo > j.LastbuildFilterInterval.Duration.Seconds() {
+				log.Printf("D! Last job too old, last %s build was %v seconds ago", jobName, buildSecAgo)
+				continue
+			}
+		}
+
+		buildIds, err := job.GetAllBuildIds()
+		sort.Sort(byBuildNumber(buildIds))
+
+		if err != nil {
+			log.Printf("E! Error retrieving all builds from job \"%s\": %s", jobName, err.Error())
+			return err
+		}
+
+		for _, buildId := range buildIds {
+			build, err := job.GetBuild(buildId.Number)
+			log.Printf("D! Reading data from job %s build %v", jobName, buildId.Number)
+			if err != nil {
+				log.Printf("E! Error retrieving build from job \"%s\": %s", jobName, err.Error())
+				return err
+			}
+
+			// ignore if build is ongoing
+			if build.IsRunning() {
+				log.Printf("D! Ignore running build on %s, build %v", jobName, buildId.Number)
+				continue
+			}
+
+			// stop if build is too old
+			if (j.LastbuildFilterInterval != internal.Duration{Duration: 0}) {
+				buildSecAgo := time.Now().Sub(build.GetTimestamp()).Seconds()
+
+				if time.Now().Sub(build.GetTimestamp()).Seconds() > j.LastbuildFilterInterval.Duration.Seconds() {
+					log.Printf("D! Job %s build %v too old (%v seconds ago), skipping to next job", jobName, buildId.Number, buildSecAgo)
+					break
+				}
+			}
+
+			tags := map[string]string{"job_name": jobName, "result": build.GetResult()}
+			fields := make(map[string]interface{})
+			fields["duration"] = build.GetDuration()
+			fields["result_code"] = mapResultCode(build.GetResult())
+
+			acc.AddFields(measurement, fields, tags, build.GetTimestamp())
```

Comment:
Will a completed build ever change?  I wonder if we should make any effort to ignore builds we have already collected?

Reply:
AFAIK a completed build should not change. How do you see this could be implemented? Maybe keeping a list of last build ID & timestamp collected per jenkins instance?

Reply:
Perhaps we could store the last reported build per job in a `map[string]int64`? If there is a gap we could record the last build before the gap for simplicity:

lastCollectedBuild = 1
```
build status
1     success
2     running
3     success
```

We would probably need to clear the previous map each Gather to remove Jobs that no longer exist.